### PR TITLE
pipe jurisdiction_id through our system from ingest -> db

### PR DESCRIPTION
**Pull request by terinpw**

### CODE REVIEW

File: recidiviz/persistence/converter/person.py [hunk 1]
```diff
@@ -46,6 +46,10 @@ def copy_fields_to_builder(person_builder, proto, metadata):
     # TODO(769): Store place_of_residence only if it can be anonymized
     new.place_of_residence = None
 
+    new.jurisdiction_id = fn(normalize, 'jurisdiction_id', proto)
+    if not new.jurisdiction_id:
```
Comment (zacharylawrence) [inline recidiviz/persistence/converter/person.py]: Can we use the `default` parameter on `fn`?

File: recidiviz/utils/regions.py [hunk 2]
```diff
@@ -128,11 +129,8 @@ def get_region_manifest(region_code: str) -> Dict[str, Any]:
             as region_manifest:
         yaml_dict = yaml.load(region_manifest)
         # TODO(#386): Require jurisdiction_id once supported in codebase
-        try:
-            yaml_dict.pop('jurisdiction_id')
-        except KeyError:
-            pass
-
+        if 'jurisdiction_id' not in yaml_dict.keys():
+            yaml_dict['jurisdiction_id'] = ''
```
Comment (zacharylawrence) [inline recidiviz/utils/regions.py]: As discussed offline, let's go ahead and fill in each `manifest.yaml` with a `jurisdiction_id` so we don't have to write an empty string to the database.

File: recidiviz/persistence/converter/person.py (hunk 1 above)
Comment (terinpw) [reply recidiviz/persistence/converter/person.py]: Good look.

File: recidiviz/ingest/scrape/base_scraper.py [hunk 3]
```diff
@@ -248,9 +248,10 @@ def _generic_scrape(self, request: QueueRequest):
                 logging.info('Last seen time of person being set as: %s',
                              request.scraper_start_time)
                 metadata = IngestMetadata(self.region.region_code,
-                                          request.scraper_start_time,
-                                          self.get_enum_overrides())
-                if self.BATCH_WRITES:
+                                          self.region.jurisdiction_id,
+                                      request.scraper_start_time,
```
Comment (jessex) [inline recidiviz/ingest/scrape/base_scraper.py]: Are these three de-dents correct?

File: recidiviz/ingest/models/ingest_info.proto [hunk 4]
```diff
@@ -42,6 +42,7 @@ message Person {
     optional string full_name = 10;
     optional string middle_names = 11;
     optional string name_suffix = 13;
+    optional string jurisdiction_id = 14;
```
Comment (colincadams) [inline recidiviz/ingest/models/ingest_info.proto]: Can you make sure you have `mypy-protobuf` installed (it's in the --dev packages) and then re run the proto compiler including the `--mypy_out flag`:
`protoc recidiviz/ingest/models/ingest_info.proto  --python_out . --mypy_out .`

This makes sure that the type hints also get updated. I'll add a comment to the proto with this info and also look into an ifttt linter.

File: recidiviz/common/ingest_metadata.py [hunk 5]
```diff
@@ -27,9 +27,12 @@
 class IngestMetadata(DefaultableAttr):
```
Comment (colincadams) [inline recidiviz/common/ingest_metadata.py]: what do you think about requiring keywords on init? (`kw_only=True`)

File: recidiviz/persistence/persistence.py [hunk 6]
```diff
@@ -63,15 +63,16 @@
 ERROR_THRESHOLD = 0.5
 
 
-def infer_release_on_open_bookings(region, last_ingest_time, custody_status):
+def infer_release_on_open_bookings(
+        region_code, last_ingest_time, custody_status):
```
Comment (colincadams) [inline recidiviz/persistence/persistence.py]: add type `str` to ensure it's the `region_code`?

File: recidiviz/ingest/scrape/base_scraper.py (hunk 3 above)
Comment (terinpw) [reply recidiviz/ingest/scrape/base_scraper.py]: I have no idea how these happened -- but good catch!

File: recidiviz/persistence/persistence.py (hunk 6 above)
Comment (terinpw) [reply recidiviz/persistence/persistence.py]: Yup! Done

File: recidiviz/common/ingest_metadata.py (hunk 5 above)
Comment (terinpw) [reply recidiviz/common/ingest_metadata.py]: I'm into that -- but I would say let's do it in a follow up PR, as it's pretty widely used.

File: recidiviz/ingest/models/ingest_info.proto (hunk 4 above)
Comment (terinpw) [reply recidiviz/ingest/models/ingest_info.proto]: Ooh didn't know about this. Done!

File: recidiviz/utils/regions.py (hunk 2 above)
Comment (terinpw) [reply recidiviz/utils/regions.py]: Agreed, will hold off on this PR until we've done that.

**Pull request merged**